### DISCUSSION TITLE
socket_dynamic: IKE not listening on UDP port 500 or 4500

### DISCUSSION THREAD

**Discussion by krishnamurthyvd** [strongswan/strongswan#993] [Q&A]
I have built strongswan 5.7.1 with 
./configure --sysconf=/etc --enable-openssl --enable-kernel-pb --enable-socket-dynamic --disable-socket-default  --prefix=/usr --enable-error-notify --enable-monolithic --sysconfdir=/etc --with-piddir=/etc/ipsec.d/run

and installed the binaries. But if start a client and server on 2 name spaces, I don't see tunnels getting established. Problem seems to be that server side (which has start_action as none) does not listen on port 500 or 4500.  I see the below error on server side tcpdump. So client side does not get any response from server. 

19:31:48.269364 IP 11.11.11.11 > 11.11.11.12: ICMP 11.11.11.11 udp port 500 unreachable, length 556
19:32:01.229603 IP 11.11.11.12.500 > 11.11.11.11.500: isakmp: parent_sa ikev2_init[I]
19:32:01.229646 IP 11.11.11.11 > 11.11.11.12: ICMP 11.11.11.11 udp port 500 unreachable, length 556

I have tried local_port and remote_port both with 500 and 4500

Is there any configuration that is needed for use with socket_dynamic option? we need to use IKE with non default remote ports (ports other than 4500 and 500)

**Comment by tobiasbrunner** [comment]
> I have built strongswan 5.7.1 

Why would you use such an old version when building from sources?

>  --enable-kernel-pb

What's that?

> Problem seems to be that server side (which has start_action as none) does not listen on port 500 or 4500.

The _socket-dynamic_ plugin does not open listening ports when the daemon starts. It's intended for clients to initiate connections from specific source ports.

> we need to use IKE with non default remote ports (ports other than 4500 and 500)

Then just use the _socket-default_ plugin and change the port (see [here](https://docs.strongswan.org/strongswan-docs/5.9/features/natTraversal.html#_custom_server_ports)).
